File: 3_Algorithms/julia/2_linear_search.jl
# 2) Linear search
# Run: julia 2_linear_search.jl

"""Return the index of target in arr, or -1 if not found.

Note: Julia uses 1-based indexing.
"""
function linear_search(arr, target)
    for index in eachindex(arr)
        if arr[index] == target
            return index
        end
    end
    return -1
end

function main()
    numbers = [4, 2, 7, 1, 9, 3]
    target = 9

    result = linear_search(numbers, target)

    if result != -1
        println("Element $(target) found at index $(result)")
    else
        println("Element $(target) not found in the list")
    end
end

if abspath(PROGRAM_FILE) == @__FILE__
    main()
end

File: 3_Algorithms/javascript/2_linear_search.js
// 2) Linear search
// Run: node 2_linear_search.js

function linearSearch(arr, target) {
  for (let index = 0; index < arr.length; index += 1) {
    if (arr[index] === target) {
      return index; // Target found
    }
  }
  return -1; // Target not found
}

function main() {
  const numbers = [4, 2, 7, 1, 9, 3];
  const target = 9;

  const result = linearSearch(numbers, target);

  if (result !== -1) {
    console.log(`Element ${target} found at index ${result}`);
  } else {
    console.log(`Element ${target} not found in the list`);
  }
}

main();
